Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -802,9 +802,57 @@ auto WebsiteDataStore::computeWebProcessAccessTypeForDataRemoval(OptionSet<Websi
 {
     if (dataTypes.contains(WebsiteDataType::MemoryCache))
         return ProcessAccessType::OnlyIfLaunched;
+
+    if (dataTypes.contains(WebsiteDataType::ResourceLoadStatistics))
+        return ProcessAccessType::OnlyIfLaunched;
+
     return ProcessAccessType::None;
 }
 
+static WebsiteDataStore::ProcessAccessType computeWebProcessAccessTypeForDataRemovalWithRecords(OptionSet<WebsiteDataType> dataTypes)
+{
+    // FIXME: We currently don't remove resource load statistics in web process for origin data removal as TestRunner might dead lock.
+    if (dataTypes.contains(WebsiteDataType::MemoryCache))
+        return WebsiteDataStore::ProcessAccessType::OnlyIfLaunched;
+
+    return WebsiteDataStore::ProcessAccessType::None;
+}
+
+HashSet<WebCore::ProcessIdentifier> WebsiteDataStore::activeWebProcesses(bool workerProcessCanBeActive) const
+{
+    HashSet<WebCore::ProcessIdentifier> identifiers;
+    for (Ref processPool : WebProcessPool::allProcessPools()) {
```

Comment:
Why can't we use this WebsiteDataStore function directly?
```
const WeakHashSet<WebProcessProxy>& processes() const { return m_processes; }
```

Reply:
`m_processes` only contains process that has page, so not including worker process, see `WebProcessProxy::updateRegistrationWithDataStore()`

Reply:
This think this needs a code comment then. Otherwise someone will refactor and break this in the future.